### ISSUE TITLE
Add icon for amenity=school

### ISSUE THREAD

**Issue by al--** [gravitystorm/openstreetmap-carto#120]
Is the green school icon ![education_school p 16](https://f.cloud.github.com/assets/1458456/979279/5c0c5340-06f0-11e3-9026-0e19ac2dd0f4.png) missing? Why?

Brian also did icons for kindergarten or university...

/al

**Comment by althio** [comment]
We should not assume that every school has a name or is tagged with a name.
from taginfo:
amenity=school with name=* :: 526 000 (83%)
amenity=school without name=* :: 106 000

An icon is useful for schools without name.
An icon is needed IMO for school on nodes, and nice to have on areas.
An icon takes less space than a name and could be rendered earlier or with less clash from other icons and labels.
[also a valid comment for #660]

The label name on the other hand should be rendered very late.
Absolute zoom levels are not enough since rendering icons and/or labels depends on the size (area) and on the density of features (clashes).

Lastly, when the school is mapped with details, the icon rendered is playground. Confusing for people looking for an public playground and confusing for people looking for the school (school can be used as any landmark and locally well-known place for orientation).
The school icon or label should somehow have more weight than the playground-in-a-school icon.

**Comment by daganzdaanda** [comment]
+1 for adding an icon for schools. 
In some areas of Africa, I've seen lots of schools that are just nodes (maybe imported). These don't always have a name. I'd rate schools important enough that the icon should be visible at z>=14 like hospitals.

**Comment by dispiste** [comment]
+ 1 for adding an icon for schools. Copy-pasting some reasons from talk mailing list:

"At least in Spain, they are very
important buildings from several points of view:
- social relevance (they congregate a very important amount of people
on a daily basis)
- for orientation (they are prominent and characteristic buildings
that can be used as reference locations)
- traffic influence (they often cause traffic jams or crowds in their
surrounding areas on opening and closing hours)"

Regarding the icon to use, I've found in Google a number of topics that work for successfully representing a school: blackboard, child(en), teacher, backpack or school bag, pencil, notebook, compass, protractor, ruler, abacus...

I specially the following combinations:
- child sitting on a school desk
- teacher writing on a blackboard
- child writing on a blackboard
- compass + protractor + ruler

**Comment by kocio-pl** [comment]
My first set of drafts:
1) ![school-14-1](https://cloud.githubusercontent.com/assets/5439713/9788491/81e8dcfa-57ca-11e5-805c-682eff4ea794.png)
2) ![school-14-2](https://cloud.githubusercontent.com/assets/5439713/9788566/0ea6740e-57cb-11e5-81bf-f518c8952af2.png)
3) ![school-14-3](https://cloud.githubusercontent.com/assets/5439713/9788572/17a5e8f0-57cb-11e5-95a3-48c68a336b7a.png)

**Comment by rmikke** [comment]
I'd say #2 is most readable.

**Comment by matkoniecz** [comment]
1, 2 looks like a subway train on tracks for me.

3, 4 are preferable (but I am still not convinced that adding icon here is a good idea)

**Comment by kocio-pl** [comment]
I also think 1, 2 may be too similar to train, that's why I think a letter is important.

Regarding the icon itself - I think school is very similar to hospital, it's just not health-related, so the color should be different, and also similar to other amenities like museum or police station. I think [this comment](https://github.com/gravitystorm/openstreetmap-carto/issues/120#issuecomment-70403943) summarizes why it's a good thing (+ language independence of course - try to find school in China for example).

**Comment by polarbearing** [comment]
+1 for shape 3 and 4. 
As for the colour, we might consider this together with other "Societal amenities" as discussed in #1624.

Should amenity=kindergarten also get an icon once school gets one?

**Comment by kocio-pl** [comment]
We have also university and college amenities and we can go in many directions. All those POIs are similar, so:
1) we can give them all the same icon, because now they are rendered the same (blackboard=education)
2) we can give blackboard for lower levels of education (kindergarten + school) and create hat icon for higher levels (university + college)
3) we could also try to make a special icon for kindergarten (different than playground)

Of course it's not sure that those icons would be accepted, but it's good to know if we want them - maybe one for all would be enough and it's not worth even trying to draft more?

**Comment by HolgerJeromin** [comment]
many pois without a name i found were duplicate in this location. 
For example a lone area with school and a building or node in it with the details. 
or an area with alle details and the building itself has a amenity=school tag.

This is valid only for good mapped places.

**Comment by polarbearing** [comment]
@kocio-pl - my preference would be separate icons. For Univ/college, a college hat is often used, for schools the blackboard, and for kindergarten I was thinking about reusing the shop=toy icon just in different colour (however this is currently a rocking horse, that might not be optimal for kindergarten).

@HolgerJeromin - duplicate tagging is a tagging error (that becomes more visible with the icon). Once the campus is mapped as amenity, it should be removed from the building, only building=school should remain. Exception might be a campus shared by separate schools that cannot be divided horizontally.

**Comment by kocio-pl** [comment]
So we can start with blackboard for schools and once we know if the other icons are good enough, we can add them later (or just expand the scope of the blackboard icon in case these attempts fail).

What's your ideas for kindergarten to test? Remember we have only 14x14 px in one color, so the simplest and more characteristic shape, the better.

**Comment by HolgerJeromin** [comment]
perhaps the new community_centre icon with two people can be modified to have one big and one small person.

**Comment by kocio-pl** [comment]
You talk about the idea for kindergarten icon - or maybe more than that (lower level education or maybe education in general)?

**Comment by HolgerJeromin** [comment]
Kindergarten. 
School could be (in Germany) to the age of 19 years. There students are taler than the teachers (ok, would result in the same icon :-)
Joking aside. The metaphor: kindergarten is more caring (big/small person) and school is more teaching (with an "A") works IMO.

**Comment by kocio-pl** [comment]
University draft:
1) ![university-14-12](https://cloud.githubusercontent.com/assets/5439713/10074373/d0336118-62cf-11e5-897a-cbe199194c63.png)

**Comment by HolgerJeromin** [comment]
i like both 1). Kindergarden is really good IMO.
2) is for me not clearly human, but could be a gas cylinder :-)

**Comment by kocio-pl** [comment]
I like it less, because visual details are not so clear, but it's still usable.

**Comment by dispiste** [comment]
I like the university icon and also the 1) kindergarten icon.

Regarding the school icon, it is not evident for me that it represents a blackboard (nor a school). What does the A stand for?

I attach some icons that may provide some inspiration for school. I have no idea on icon design, so they probably have to be re-worked a bit to look more professional. They are based on your kindergarten icon:

1) ![desk](https://cloud.githubusercontent.com/assets/1564127/10098969/dfdbe822-6385-11e5-82d5-7b45068c0ae2.png)
2) ![desk2](https://cloud.githubusercontent.com/assets/1564127/10098976/efb1fe76-6385-11e5-86fb-14186defbcb3.png)
3) ![desk3](https://cloud.githubusercontent.com/assets/1564127/10098983/f978bee0-6385-11e5-8b2c-62cd6fac98e6.png)
4) ![desk4](https://cloud.githubusercontent.com/assets/1564127/10098995/fe69ce26-6385-11e5-8aba-5f4777c9dffe.png)
5) ![blackboard1](https://cloud.githubusercontent.com/assets/1564127/10099002/0d1a10de-6386-11e5-9269-2debc54f8bd2.png)
6) ![blackboard0](https://cloud.githubusercontent.com/assets/1564127/10099011/164776ec-6386-11e5-8e2c-f8f5ae865bbd.png)

**Comment by kocio-pl** [comment]
Thanks a lot! We need more contributors to make healthy community, no matter what kind of input. I like 5) the most, because it's not too light. Do you have SVG for that or is it only a raster draft?

**Comment by dispiste** [comment]
It is only a pixel-drawn raster draft.

**Comment by dispiste** [comment]
I've now created some SVG versions of 5), but I'm not sure they are clear enough when rendered to 14x14:

5.1) ![osm-school0](https://cloud.githubusercontent.com/assets/1564127/10100730/fc9a78bc-6393-11e5-9388-788cc6c2ae69.png) (osm-school0.svg)
5.2) ![osm-school2](https://cloud.githubusercontent.com/assets/1564127/10100705/e303efbe-6393-11e5-91ac-040c4183ad0e.png) (osm-school1.svg)
5.3) ![osm-school3](https://cloud.githubusercontent.com/assets/1564127/10100708/ec9b3780-6393-11e5-8d06-9c17cf6893b1.png)  (osm-school2.svg)
5.4) ![osm-school](https://cloud.githubusercontent.com/assets/1564127/10100700/dd3c6656-6393-11e5-80df-40b07656206a.png)  (osm-school.svg)

**Comment by polarbearing** [comment]
Thanks @dispiste for the contribution. Though the "blackboard against the wall" is the more typical real-world setup, the "blackboard on the easel" appears to me more recognisable as a school in the small icon format.

**Comment by kocio-pl** [comment]
Could you share the osm-school.svg file (5.4)? 

It's most probably problem of pixel aligning - if we make the proper scalable icon, this is not always possible, but we can try making bigger gap for example.

**Comment by dispiste** [comment]
Thanks @kocio-pl for your efforts.
SVG 5.4 is available here:
https://github.com/dispiste/dispiste.github.io/blob/master/osm/symbols/osm-school.svg

I've uploaded all of them to:
https://github.com/dispiste/dispiste.github.io/blob/master/osm/symbols/

**Comment by dispiste** [comment]
Two more variation added:
5.5) ![osm-school5](https://cloud.githubusercontent.com/assets/1564127/10104860/1978e4d8-63ac-11e5-8b1d-3841ea39deb0.png)
5.6) ![osm-school6](https://cloud.githubusercontent.com/assets/1564127/10105400/cb221022-63ae-11e5-8b00-fc9c65bcbaa2.png)


Available on:
https://github.com/dispiste/dispiste.github.io/blob/master/osm/symbols/osm-school5.svg
https://github.com/dispiste/dispiste.github.io/blob/master/osm/symbols/osm-school6.svg

**Comment by kocio-pl** [comment]
Are we still interested in icons for educational institutions? If yes, I could provide the code (once we decide which icon would be the best for school), otherwise it should be closed.

**Comment by daganzdaanda** [comment]
I still would like to see an icon for schools. Universities, Kindergartens / Childcare facilities would make sense, too.

**Comment by kocio-pl** [comment]
IIRC university and kindergarten icons are ready and I guess the best school icons left to choose from are:
7) ![school-14-9](https://cloud.githubusercontent.com/assets/5439713/10101045/5c8257ca-6396-11e5-9a5b-3c3d1bcd1307.png)
5.6) ![](https://cloud.githubusercontent.com/assets/1564127/10105400/cb221022-63ae-11e5-8b00-fc9c65bcbaa2.png)

**Comment by polarbearing** [comment]
Still prefer 7) typus, see comment 25 Sep 2015. 

For Kindergarten, could we have the icon more oblique, so that the children appear running? Similar dynamic was brought into the related [German traffic sign](https://upload.wikimedia.org/wikipedia/commons/e/eb/Zeichen_136-10.svg) a while ago.

**Comment by dispiste** [comment]
Although I feel we have not found the perfect icon for schools yet, I think
the proposed icons are better than nothing, so please go for it. It will be
really useful.

On 21 February 2016 at 23:26, polarbearing <notifications@github.com> wrote:

> Still prefer 7) typus, see comment 25 Sep 2015.
>
> For Kindergarten, could we have the icon more oblique, so that the
> children appear running? Similar dynamic was brought into the related German
> traffic sign
> <https://upload.wikimedia.org/wikipedia/commons/e/eb/Zeichen_136-10.svg>
> a while ago.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gravitystorm/openstreetmap-carto/issues/120#issuecomment-186933691>
> .
>



-- 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   César Martínez Izquierdo
   GIS developer
   -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -
   SCOLAB: http://www.scolab.es
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

**Comment by kocio-pl** [comment]
@polarbearing It was meant to be adult with child, so I think static version is better to not be confused with school kids.

**Comment by polarbearing** [comment]
Yes I know the intended meaning. My concern is to differentiate clearly against the two people in the community_centre, thus having them running in the kindergarten icon would help. There is no visual overlap with the blackboard proposed here for school. Further, amenity=kindergarten is increasingly used for afternoon care of primary school kids (de: Hort). Thus, two running kids of slightly different size would be fine for me.

**Comment by daganzdaanda** [comment]
University: yep, thats good! 

Kindergarten: Running children might be a bit more intuitive than a static grown-up and child. What do people outside of Germany think? Another option might be some toy blocks, but I believe that would be harder to read at the small size.

Schools: I like both icons 7 and 5.6, don't have a real preference.

**Comment by jragusa** [comment]
I can test rendering if you're still interested to show icons for this amenities. I just need links to svg icons

**Comment by jragusa** [comment]
@kocio-pl ping ?

**Comment by kocio-pl** [comment]
The PR was created but closed (#2361) because I don't know how to deal with label scaling in such cases.

**Comment by jragusa** [comment]
Oups, sorry for the disturbance

**Comment by kocio-pl** [comment]
No problem. :smile:

**Comment by jeisenbe** [comment]
See #3856 for examples of aerodrome labels and icons which are adjusted to different sizes at different zoom levels.

**Comment by sun-geo** [comment]
I found icons for schools using osm.org (on left panel), the iD editor and JOSM, see below . 

![school icon](https://user-images.githubusercontent.com/47290585/98316544-dca40380-1fda-11eb-8e8d-e5c2b81af51d.png)

I was wondering if the osm-carto map might also show it.

Then i found this long standing discussion and hope it will be implemented one day. I don't understand the mentioned scaling issue. Could it maybe just work the same way which is implemented for the other similar yellow area object-icons (`amenity=community_centre` / `amenity=social_facility` / `amenity=arts_centre`)?

![image](https://user-images.githubusercontent.com/47290585/98318544-18d96300-1fdf-11eb-90d0-01b0ce5e7e97.png)

**Comment by Circeus** [comment]
> I don't understand the mentioned scaling issue. 

Whether or not the names of some features such as parks and Cemetary are visible is beholden only to the amount of available space to draw the name rather than the zoom level. A feature with label scaling that is large enough can have its name shown at _any_ zoom level as long as it is traced and there's room to fit the name. This is especially noticeable with national parks or military bases, which have names as far out as zoom 8 (they are no longer traced at zoom 7).

By contrast, anything with an associated icon is strictly limited by zoom levels. [Calverton National Cemetery](https://www.openstreetmap.org/way/50029602) has a name visible as early as zoom 13. By contrast, no hospital _anywhere in the world_ has a name visible before zoom 16. Their icons don't show up at zoom 14. Scaled labels are also much less restricted by collision limits with nearby features.

So basically a choice has to be made between name scaling and having an icon. I think most people prefer scaled labels over icon issues.
